src/wp-includes/option.php

```diff
@@ -2924,3 +3029,29 @@ function filter_default_option( $default_value, $option, $passed_default ) {
 
 	return $registered[ $option ]['default'];
 }
+
+/**
+ * Converts the values that trigger autoloading from the options table to a string or an array.
+ *
+ * The values that trigger autoloading are filtered through the 'wp_autoload_values_to_autoload' hook, allowing customization.
+ *
+ * @since 5.6.0
+ *
+ * @param bool $as_string Optional. Whether to return the values as a string or an array. Default false.
+ *
+ * @return string|array The values that trigger autoloading.
+ */
+function wp_autoload_values_to_autoload( $as_string = false ) {
+	/**
+	 * Filter: allows the values use to trigger a value to be autoloaded from the options table
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param array $default values to trigger autoloading option.
+	 *
+	 * @return array $values to trigger autoloading option.
+	 */
+	$values = apply_filters( 'wp_autoload_values_to_autoload', array( 'yes', 'on', 'auto-on', 'auto' ) );
+
+	return ( $as_string ) ? "'" . implode( "', '", $values ) . "'" : $values;
+}
```

Comment:
Let's remove this parameter. A boolean parameter to change what a function does is an antipattern and violates the single responsibility principle. We should either introduce a dedicated function to get this as a string, or simply handle it inline where needed.

Also, why is there a filter? This is a WordPress core internal detail that shouldn't be filterable, at least not unless/until there is a real reason to provide a filter.
```suggestion
/**
 * Returns the values that trigger autoloading from the options table.
 *
 * @since 6.6.0
 *
 * @return array The values that trigger autoloading.
 */
function wp_autoload_values_to_autoload() {
	return array( 'yes', 'on', 'auto-on', 'auto' );
}
```